test/unit/specs/components/ActivityCollector/attachClickActivityCollector.spec.js

```diff
@@ -110,12 +105,12 @@ describe("ActivityCollector::attachClickActivityCollector", () => {
 
   it("handles errors thrown in sendEvent", () => {
     const error = new Error("Network Error");
-    eventManager.createEvent = () => {
-      return {
+    eventManager = jasmine.createSpyObj("eventManager", {
+      createEvent: {
         isEmpty: () => false
-      };
-    };
-    spyOn(eventManager, "sendEvent").and.returnValue(Promise.reject(error));
+      },
+      sendEvent: Promise.reject(error)
+    });
```

Comment:
Rather than recreating the full mock, you should be able to do:

```
eventManager.createEvent.and.returnValue({
  isEmpty: () => false
});
```

The same can be applied to the previous test.